llvm/lib/Target/RISCV/GISel/RISCVRegisterBankInfo.cpp

```diff
@@ -269,6 +287,9 @@ RISCVRegisterBankInfo::getInstrMapping(const MachineInstr &MI) const {
   case TargetOpcode::G_ZEXT:
   case TargetOpcode::G_SEXTLOAD:
   case TargetOpcode::G_ZEXTLOAD:
+    if (MRI.getType(MI.getOperand(0).getReg()).isVector())
+      return getInstructionMapping(DefaultMappingID, /*Cost=*/1,
```

Comment:
This does regbankselect for many more opcodes than just G_ADD and G_SUB. Can you please change to only map to vector for the instructions you mention in PR title/description + include regbankselect tests for.